[CI] EsqlClientYamlIT test {p0=esql/61_enrich_ip/IP strings} failing

**Build Scans:**
- [elasticsearch-periodic #4786 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/oyorwichebtf4)
- [elasticsearch-pull-request #40468 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/4nadaxpo6lh4a)
- [elasticsearch-pull-request #40468 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/mvribf3qhi3ve)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.13.4#yamlRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT" -Dtests.method="test {p0=esql/61_enrich_ip/IP strings}" -Dtests.seed=428BBA97DEF79736 -Dtests.bwc=true -Dtests.locale=csw-Cans-CA -Dtests.timezone=US/Pacific -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Desql%2F61_enrich_ip%2FIP%20strings%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [esql/61_enrich_ip:112]: expected [2xx] status code but api [esql.query] returned [500 Internal Server Error] [{"error":{"root_cause":[{"type":"null_pointer_exception","reason":"Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.ordinal()\" because \"inputDataType\" is null","stack_trace":"org.elasticsearch.ElasticsearchException$1: Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.ordinal()\" because \"inputDataType\" is null\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:701)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:629)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.rest.RestResponse.build(RestResponse.java:184)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:143)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:114)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:55)\n\tat org.elasticsearch.xpack.esql.action.EsqlResponseListener.lambda$wrapWithLogging$1(EsqlResponseListener.java:173)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListener$2.onFailure(ActionListener.java:179)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.rest.action.RestCancellableNodeClient$1.onFailure(RestCancellableNodeClient.java:96)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:324)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:214)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:324)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$execute$1(ComputeService.java:147)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:186)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:191)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:324)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:324)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.support.RefCountingListener.finish(RefCountingListener.java:127)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:151)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.support.RefCountingListener$1.onResponse(RefCountingListener.java:146)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onResponse(ActionListenerImplementations.java:269)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListener$3.onResponse(ActionListener.java:314)\n\tat org.elasticsearch.xpack.esql.plugin.ComputeService.lambda$startComputeOnDataNodes$13(ComputeService.java:298)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.AbstractRefCounted$1.closeInternal(AbstractRefCounted.java:118)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.AbstractRefCounted.decRef(AbstractRefCounted.java:70)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.support.RefCountingRunnable.close(RefCountingRunnable.java:112)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.Releasables$4.close(Releasables.java:159)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.Releasables.close(Releasables.java:34)\n\tat org.elasticsearch.base@8.13.4/org.elasticsearch.core.Releasables.closeExpectNoException(Releasables.java:56)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations$2.run(ActionListenerImplementations.java:49)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations$RunAfterActionListener.onFailure(ActionListenerImplementations.java:280)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:324)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:53)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleException(TransportService.java:1751)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1475)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1609)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.transport.TransportService$DirectResponseChannel$2.doRun(TransportService.java:1589)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.13.4/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1583)\nCaused by: java.lang.NullPointerException: Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.ordinal()\" because \"inputDataType\" is null\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService.termQueryList(AbstractLookupService.java:185)\n\tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:82)\n\tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:42)\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService.doLookup(AbstractLookupService.java:328)\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:433)\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:428)\n\tat org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:76)\n\tat org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:289)\n\tat org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:302)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.lang.Thread.run(Thread.java:1575)\n"}],"type":"null_pointer_exception","reason":"Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.ordinal()\" because \"inputDataType\" is null","stack_trace":"java.lang.NullPointerException: Cannot invoke \"org.elasticsearch.xpack.esql.core.type.DataType.ordinal()\" because \"inputDataType\" is null\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService.termQueryList(AbstractLookupService.java:185)\n\tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:82)\n\tat org.elasticsearch.xpack.esql.enrich.EnrichLookupService.queryList(EnrichLookupService.java:42)\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService.doLookup(AbstractLookupService.java:328)\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:433)\n\tat org.elasticsearch.xpack.esql.enrich.AbstractLookupService$TransportHandler.messageReceived(AbstractLookupService.java:428)\n\tat org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:76)\n\tat org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:289)\n\tat org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:302)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1023)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.lang.Thread.run(Thread.java:1575)\n"},"status":500}]
```

**Issue Reasons:**
- [8.x] 3 failures in test test {p0=esql/61_enrich_ip/IP strings} (2.5% fail rate in 120 executions)
- [8.x] 2 failures in step 8.13.4_bwc (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/docs/changelog/116583.yaml
+++ b/docs/changelog/116583.yaml
@@ -1,0 +1,7 @@
+pr: 116583
+summary: Fix NPE in `EnrichLookupService` on mixed clusters with <8.14 versions
+area: ES|QL
+type: bug
+issues:
+ - 116529
+ - 116544

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
@@ -45,6 +45,7 @@ import org.elasticsearch.compute.operator.lookup.EnrichQuerySourceOperator;
 import org.elasticsearch.compute.operator.lookup.MergePositionsOperator;
 import org.elasticsearch.compute.operator.lookup.QueryList;
 import org.elasticsearch.core.AbstractRefCounted;
+import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.RefCounted;
 import org.elasticsearch.core.Releasable;
 import org.elasticsearch.core.Releasables;
@@ -185,7 +186,7 @@ abstract class AbstractLookupService<R extends AbstractLookupService.Request, T 
         return switch (inputDataType) {
             case IP -> QueryList.ipTermQueryList(field, searchExecutionContext, (BytesRefBlock) block);
             case DATETIME -> QueryList.dateTermQueryList(field, searchExecutionContext, (LongBlock) block);
-            default -> QueryList.rawTermQueryList(field, searchExecutionContext, block);
+            case null, default -> QueryList.rawTermQueryList(field, searchExecutionContext, block);
         };
     }
 
@@ -459,6 +460,7 @@ abstract class AbstractLookupService<R extends AbstractLookupService.Request, T 
     abstract static class TransportRequest extends org.elasticsearch.transport.TransportRequest implements IndicesRequest {
         final String sessionId;
         final ShardId shardId;
+        @Nullable
         final DataType inputDataType;
         final Page inputPage;
         final List<NamedExpression> extractFields;

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
@@ -461,7 +461,7 @@ abstract class AbstractLookupService<R extends AbstractLookupService.Request, T 
         final String sessionId;
         final ShardId shardId;
         /**
-         * For mixed clusters with nodes <8.14, this will be null.
+         * For mixed clusters with nodes < 8.14, this will be null.
          */
         @Nullable
         final DataType inputDataType;

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
@@ -460,6 +460,9 @@ abstract class AbstractLookupService<R extends AbstractLookupService.Request, T 
     abstract static class TransportRequest extends org.elasticsearch.transport.TransportRequest implements IndicesRequest {
         final String sessionId;
         final ShardId shardId;
+        /**
+         * For mixed clusters with nodes <8.14, this will be null.
+         */
         @Nullable
         final DataType inputDataType;
         final Page inputPage;

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java
@@ -461,7 +461,7 @@ abstract class AbstractLookupService<R extends AbstractLookupService.Request, T 
         final String sessionId;
         final ShardId shardId;
         /**
-         * For mixed clusters with nodes < 8.14, this will be null.
+         * For mixed clusters with nodes &lt;8.14, this will be null.
          */
         @Nullable
         final DataType inputDataType;

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java
@@ -127,9 +127,9 @@ public class EnrichLookupService extends AbstractLookupService<EnrichLookupServi
             TaskId parentTaskId = TaskId.readFromStream(in);
             String sessionId = in.readString();
             ShardId shardId = new ShardId(in);
-            DataType inputDataType = DataType.fromTypeName(
-                (in.getTransportVersion().onOrAfter(TransportVersions.V_8_14_0)) ? in.readString() : "unknown"
-            );
+            DataType inputDataType = (in.getTransportVersion().onOrAfter(TransportVersions.V_8_14_0))
+                ? DataType.fromTypeName(in.readString())
+                : null;
             String matchType = in.readString();
             String matchField = in.readString();
             Page inputPage;

